synology_api/auth.py

```diff
@@ -92,28 +92,21 @@ def request_data(self, api_name, api_path, req_param, method=None, response_json
             if isinstance(v, bool):
                 req_param[k] = str(v).lower()
 
-        if method is None:
-            method = 'get'
-
         req_param['_sid'] = self._sid
 
-        if method is 'get':
-            url = ('%s%s' % (self._base_url, api_path)) + '?api=' + api_name
+        url = ('%s%s' % (self._base_url, api_path)) + '?api=' + api_name
+        # checking and handling HTTP-Method (perform a request)
+        if method.lower() is 'get' or method is None:

```

Comment:
Default  'method' must be specified as all function do not specify what method will be used when 'request_data()' is called, 'None' will pass through the conditional rising errors like 'NoneType' object has no attribute 'lower'